File: walletsdk/src/main/java/leaf/prod/walletsdk/model/order/OrderStatus.java
/**
 * Created with IntelliJ IDEA.
 * User: kenshin wangchen@loopring.org
 * Time: 2018-11-16 11:35 AM
 * Cooperation: loopring.org 路印协议基金会
 */
package leaf.prod.walletsdk.model.order;

import java.io.Serializable;

import android.content.Context;

import com.google.gson.annotations.SerializedName;

import leaf.prod.walletsdk.R;

public enum OrderStatus implements Serializable {

    @SerializedName(value = "STATUS_PENDING_ACTIVE")
    PENDING(R.string.order_pending),

    @SerializedName(value = "STATUS_NEW", alternate = {"STATUS_PENDING", "STATUS_PARTIALLY_FILLED"})
    OPENED(R.string.order_open),

    @SerializedName("STATUS_COMPLETELY_FILLED")
    FINISHED(R.string.order_completed),

    @SerializedName(value = "STATUS_SOFT_CANCELLED_BY_USER", alternate = {"STATUS_ONCHAIN_CANCELLED_BY_USER", "STATUS_ONCHAIN_CANCELLED_BY_USER_TRADING_PAIR", "STATUS_SOFT_CANCELLED_BY_USER_TRADING_PAIR"})
    CANCELLED(R.string.order_cancelled),

    @SerializedName("STATUS_EXPIRED")
    EXPIRED(R.string.order_expired),

    @SerializedName("ORDER_UNKNOWN")
    UNKNOWN(R.string.order_unknown);

    private int resourceId;

    OrderStatus(int id) {
        this.resourceId = id;
    }

    public String getDescription(Context context) {
        return context.getString(resourceId);
    }
}

File: loopr-ios/Models/Order/OrderStatus.swift
//
//  OrderStatus.swift
//  loopr-ios
//
//  Created by xiaoruby on 2/5/18.
//  Copyright © 2018 Loopring. All rights reserved.
//
import UIKit

enum OrderStatus: String, CustomStringConvertible {

    case pending_active = "STATUS_PENDING_ACTIVE"
    case new = "STATUS_NEW"
    case pending = "STATUS_PENDING"
    case partially_filled = "STATUS_PARTIALLY_FILLED"
    case soft_cancelled_by_user = "STATUS_SOFT_CANCELLED_BY_USER"
    case onchain_cancelled_by_user = "STATUS_ONCHAIN_CANCELLED_BY_USER"
    case soft_cancelled_by_user_trading_pair = "STATUS_SOFT_CANCELLED_BY_USER_TRADING_PAIR"
    case onchain_cancelled_by_user_trading_pair = "STATUS_ONCHAIN_CANCELLED_BY_USER_TRADING_PAIR"
    case completely_filled = "STATUS_COMPLETELY_FILLED"
    case expired = "STATUS_EXPIRED"
    case unknown = "ORDER_UNKNOWN"

    var description: String {
        switch self {
        case .pending_active: return LocalizedString("Order Pending", comment: "")
        case .new: return LocalizedString("Order Submitted", comment: "")
        case .pending: return LocalizedString("Order Submitted", comment: "")
        case .partially_filled: return LocalizedString("Order Submitted", comment: "")
        case .soft_cancelled_by_user: return LocalizedString("Order Cancelled", comment: "")
        case .onchain_cancelled_by_user: return LocalizedString("Order Cancelled", comment: "")
        case .soft_cancelled_by_user_trading_pair: return LocalizedString("Order Cancelled", comment: "")
        case .onchain_cancelled_by_user_trading_pair: return LocalizedString("Order Cancelled", comment: "")
        case .completely_filled: return LocalizedString("Order Completed", comment: "")
        case .expired: return LocalizedString("Order Expired", comment: "")
        case .unknown: return LocalizedString("Order Unknown", comment: "")
        }
    }
}
